### PR TITLE
Convert to C API logging

### PR DESCRIPTION
**Pull request by yongtang**
This PR convert to C API logging in modular file system plugins. The `tensorflow/c/logging.h` is already available, however, the library didn't link the API (will be fixed by https://github.com/tensorflow/tensorflow/pull/44993).

For the moment this PR converts the C++ `LOG` call which will be needed very soon anyway.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow_io/core/plugins/az/az_file_system.cc [hunk 1]
```diff
@@ -31,6 +31,12 @@ limitations under the License.
 #include "storage_account.h"
 #include "storage_credential.h"
 #include "storage_errno.h"
+// TODO: Restore logging.h
+#define TF_Log(...)
+#define TF_VLog(...)
```
Comment (kvignesh1420) [inline tensorflow_io/core/plugins/az/az_file_system.cc:36]: @yongtang the `TF_VLog` macro isn't being used here, is this just to bring in the existing logging macro's ?

**Review by terrytangyuan — APPROVED** [review]
LGTM

File: tensorflow_io/core/plugins/az/az_file_system.cc (hunk 1 above)
Comment (yongtang) [reply tensorflow_io/core/plugins/az/az_file_system.cc:36]: @kvignesh1420 The `TF_VLog` is part of the C logging API in tensorflow (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/c/logging.h#L34-L36). Currently it is defined as empty as we need https://github.com/tensorflow/tensorflow/pull/44993 . Once the PR is merged those defines here can be safely removed.

Comment (kvignesh1420) [reply tensorflow_io/core/plugins/az/az_file_system.cc:36]: okay!

**Review by kvignesh1420 — APPROVED** [review]
LGTM

**Pull request merged**